### PR TITLE
Rename sms_fragments to sms_chargeable_units

### PR DESCRIPTION
**Pull request by CrystalPea**
This field caused some confusion and lots of unnecessary work to our colleague because of unclear name.

The field was named sms_fragments, where in fact the value of the field is: those sms fragments that go above free allowance multiplied by the rate multiplier.

The new name was chosen through consultation with colleagues who use billing report the most.

This change should go out together with: https://github.com/alphagov/notifications-api/pull/3430

TODO: see if we need guidance to explain column names in the billing report

### CODE REVIEW

**Review by leohemsted — APPROVED** [review]
Code looks fine pending any content review

File: app/templates/views/platform-admin/get-billing-report.html [hunk 1]
```diff
@@ -17,4 +17,22 @@ <h1 class="heading-large">
...
+    Explanation for more confusing column names in the report:
+  </p>
+  <ul class="govuk-list govuk-list--bullet">
+    <li>sms cost - cost accrued by a service through sending text messages after they run out of free allowance</li>
+    <li>sms chargeable units - sms fragments that go over free allowance multiplied by rate multiplier (so higher costs of sending international SMS are taken into consideration)</li>
+    <li>letter cost - cost accrued by a service through sending letters (number of letters * numbers of paper sheets * postage)</li>
+    <li>letter breakdown - amount and cost of letters sent by a service for each postage class</li>
+    <li> purchase order number, contact names, contact email addresses, billing reference - this data is manually filled in by us based on the information we get from services. You can help by filling it in (on Service Settings page)</li>
+
+  </ul>
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:36]:
```suggestion
<div class="bottom-gutter-3-2">
    {% call mapping_table(
      caption='Descriptions of billing report data',
      field_headings=['Name', 'Description'],
      field_headings_visible=True,
      caption_visible=False
    ) %}
      {% for message_length, charge in [
        ('sms cost', 'The total cost of text messages sent after a service has used its free allowance.'),
        ('sms chargeable units', 'The number of fragments sent after a service has used its free allowance. This number takes into account the cost multiplier for <a class="govuk-link govuk-link--no-visited-state" href="https://www.notifications.service.gov.uk/pricing#international-numbers">sending international text messages</a>.'),
        ('letter cost', 'The total cost of letters sent by a service.'),
        ('letter breakdown', 'The number of letters sent by a service, grouped by cost (sheets of paper × postage class).'),
        ('purchase order number, contact names, contact email addresses and billing reference', 'We add this data manually based on the information we get from services. You can help by adding it to the service settings page.'),
      ] %}
        {% call row() %}
          {{ text_field(message_length) }}
          {{ text_field(charge | safe) }}
        {% endcall %}
      {% endfor %}
    {% endcall %}
  </div>
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 2]
```diff
@@ -17,4 +17,22 @@ <h1 class="heading-large">
     {{ page_footer('Download report') }}
   {% endcall %}
 
+  <h2 class="heading-medium">
+    Guidance for interpreting the report
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:21]:
```suggestion
    Data included in the report
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 3]
```diff
@@ -17,4 +17,22 @@ <h1 class="heading-large">
...
+  </h2>
+  <p class="govuk-body">
+    Billing report shows data for period between start date and end date, inclusive of both dates.
+  </p>
+  <p class="govuk-body">
+    Explanation for more confusing column names in the report:
+  </p>
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:28]:
```suggestion
```

**Comment by CrystalPea** [comment]
Screenshot of the guidance on the Report page:

Screenshot 2022-01-28 at 15 39 23

File: app/templates/views/platform-admin/get-billing-report.html [hunk 4]
```diff
@@ -14,7 +15,33 @@ <h1 class="heading-large">
   {% call form_wrapper() %}
     {{ form.start_date(param_extensions={"hint": {"text": "Enter start date in format YYYY-MM-DD"}}) }}
     {{ form.end_date(param_extensions={"hint": {"text": "Enter end date in format YYYY-MM-DD"}}) }}
+    <p class="hint">Generated report will be inclusive of both dates.</p>
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:18]: Suggest we remove this and change the field names from:

* `Start date` to `First day covered by report`
* `End date` to Last day covered by report`

```suggestion
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 5]
```diff
@@ -14,7 +15,33 @@ <h1 class="heading-large">
   {% call form_wrapper() %}
     {{ form.start_date(param_extensions={"hint": {"text": "Enter start date in format YYYY-MM-DD"}}) }}
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:16]:
```suggestion
    {{ form.start_date(param_extensions={"hint": {"text": "For example, 01-01-2020"}}) }}
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 6]
```diff
@@ -14,7 +15,33 @@ <h1 class="heading-large">
...
     {{ form.end_date(param_extensions={"hint": {"text": "Enter end date in format YYYY-MM-DD"}}) }}
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:17]:
```suggestion
    {{ form.end_date(param_extensions={"hint": {"text": "For example, 01-01-2021"}}) }}
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 7]
```diff
@@ -14,7 +15,33 @@ <h1 class="heading-large">
...
+      caption='Descriptions of billing report data',
+      field_headings=['Name', 'Description'],
+      field_headings_visible=True,
+      caption_visible=False
+    ) %}
+      {% for message_length, charge in [
+        ('sms cost', 'The total cost of text messages sent after a service has used its free allowance.'),
+        ('sms chargeable units', 'The number of fragments sent after a service has used its free allowance. This number takes into account the cost multiplier for <a class="govuk-link govuk-link--no-visited-state" href="https://www.notifications.service.gov.uk/pricing#international-numbers">sending international text messages</a>.'),
+        ('letter cost', 'The total cost of letters sent by a service.'),
+        ('letter breakdown', 'The number and cost of letters sent by a service, grouped by postage.'),
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:36]:
```suggestion
        ('letter breakdown', 'The number of letters sent by a service, grouped by cost (sheets of paper × postage class).'),
```

File: app/templates/views/platform-admin/get-billing-report.html [hunk 8]
```diff
@@ -12,9 +13,35 @@ <h1 class="heading-large">
...
```
Comment (karlchillmaid) [inline app/templates/views/platform-admin/get-billing-report.html:36]:
```suggestion
        ('letter breakdown', 'The number of letters sent by a service, grouped by postage and unit cost.'),
```

**Comment by CrystalPea** [comment]
Final screenshot:

Screenshot 2022-02-01 at 16 38 48

**Pull request merged**